aiidalab_widgets_base/structures.py

```diff
@@ -224,6 +232,7 @@ def _convert_to_structure_node(self, structure):
 
         # If the input_structure trait is set to Atoms object, structure node must be created from it.
         if isinstance(structure, Atoms):
+            self._validate_and_fix_ase_cell(structure)
```

Comment:
Please put it to `_observe_input_structure`, as it receives the external data the first.

(same for below)